hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -1021,6 +1048,58 @@ coverageFailures :: TestCount -> Coverage CoverCount -> [Label CoverCount]
 coverageFailures tests (Coverage kvs) =
   filter (not . labelCovered tests) (Map.elems kvs)
 
+-- | Is true when the test coverage satisfies the specified 'Confidence'
+--   contstraint for all 'Coverage CoverCount's
+confidenceSuccess :: TestCount -> Confidence -> Coverage CoverCount -> Bool
+confidenceSuccess tests confidence =
+  let
+    assertLow :: Label CoverCount -> Bool
+    assertLow MkLabel{..} =
+      wilsonLowerBound
+        (fromIntegral $ unCoverCount labelAnnotation)
+        (fromIntegral tests)
+        (1 / fromIntegral (unConfidence confidence))
+      >= 0.9 * (unCoverPercentage labelMinimum / 100.0)
```

Comment:
This is just a tolerance - we could include that in the confidence data type. That would be clearer.

Reply:
@felixmulder, @HuwCampbell, perhaps there's something off with the above calculation? See test below:

```hs
prop_with_confidence :: Property
prop_with_confidence =
  verifiedTermination . withConfidence (10^9) . withTests 1000000 . property $ do
    number <- forAll (Gen.int $ Range.constant 1 2)
    cover 50 "number == 1" $ number == 1
    cover 50 "number == 2" $ number == 2
``

This will run forever, even though the actual coverage has been reached.

Reply:
I understand that this snippet is outdate, but to me it looks that it's around this part that the issue happens.

`confidenceSuccess` and `confidenceFailure`

Reply:
@felixmulder @HuwCampbell 🏓 🍻

Reply:
The "whole" confidence interval must be greater than the desired coverage. Here, the more tests we run just shrinks the interval from like 49.9% to 50.1% to 49.99% to 50.01%.

With more tests we just get more 9s.

Put another way, the cover query states that the _lower_ bound must be greater than the desired coverage, which it's clearly not. The 49.99999% is never getting above 50%.

Reply:
It almost looks as QuickCheck's [`checkCoverage`](https://hackage.haskell.org/package/QuickCheck-2.14.2/docs/Test-QuickCheck.html#v:checkCoverage) (when using [`cover`](https://hackage.haskell.org/package/QuickCheck-2.14.2/docs/Test-QuickCheck.html#v:cover)) is far less stringent in order to improve performance.

Here's a [side-by-side comparison](https://github.com/moodmosaic/coverage-check-example/tree/master/src/Test/Q72168364) of Hedgehog and QuickCheck with regards to this. If I use the exact same percentage, Hedgehog runs forever, while QuickCheck terminates.

@HuwCampbell, I would be very much interested in your thoughts on this. Thank you.